Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -15,6 +15,7 @@ private let KVOLoading = "loading"
 private let KVOEstimatedProgress = "estimatedProgress"
 
 private let HomeURL = "about:home"
+private let URLBeforePathRegex = NSRegularExpression(pattern: "^https?://([^/]+/)", options: nil, error: nil)!
```

Comment:
I really think we need to stop clogging up BrowserViewController with this stuff. This is driven by the SearchViewController, so it seems like a better owner.